packages/zoltan2/core/src/algorithms/zoltan/Zoltan2_AlgZoltan.hpp

```diff
@@ -596,6 +630,118 @@ class AlgZoltan : public Algorithm<Adapter>
     setCallbacksGeom(&(*adapter));
   }
 
+
+  // JAE: Not implemented yet
+  AlgZoltan(const RCP<const Environment> &env__,
+            const RCP<const Comm<int> > &problemComm__,
+            const RCP<const IdentifierAdapter<user_t> > &adapter__,
+            const RCP<const MachineRepresentation<scalar_t, part_t> > &machine__) :
+    env(env__), problemComm(problemComm__), adapter(adapter__)
+  {
+
+    if (problemComm__->getRank() == 0)
+      std::cout << "\nEnter AlgZoltan Constructor IDENTIFIER\n" << std::endl;
+
+    setMPIComm(problemComm__);
+    zoltanInit();
+    zz = rcp(new Zoltan(mpicomm));
+    setCallbacksIDs();
+
+    //setCallbacksHierTaskMapping(machine__);
+  }
+
+  AlgZoltan(const RCP<const Environment> &env__,
+            const RCP<const Comm<int> > &problemComm__,
+            const RCP<const VectorAdapter<user_t> > &adapter__,
+            const RCP<const MachineRepresentation<scalar_t, part_t> > &machine__) :
+    env(env__), problemComm(problemComm__), adapter(adapter__)
+  {
+
+    if (problemComm__->getRank() == 0)
+      std::cout << "\nEnter AlgZoltan Constructor VECTOR\n" << std::endl;
+
+    setMPIComm(problemComm__);
+    zoltanInit();
+    zz = rcp(new Zoltan(mpicomm));
+    setCallbacksIDs();
+    setCallbacksGeom(&(*adapter));
+
+    setCallbacksHierTaskMapping(machine__);
+  }
+
+  AlgZoltan(const RCP<const Environment> &env__,
+            const RCP<const Comm<int> > &problemComm__,
+            const RCP<const GraphAdapter<user_t, userCoord_t> > &adapter__,
+            const RCP<const MachineRepresentation<scalar_t, part_t> > &machine__) :
+    env(env__), problemComm(problemComm__), adapter(adapter__)
+  {
+
+    if (problemComm__->getRank() == 0)
+      std::cout << "\nEnter AlgZoltan Constructor GRAPH\n" << std::endl;
+
+    setMPIComm(problemComm__);
+    zoltanInit();
+    zz = rcp(new Zoltan(mpicomm));
+    setCallbacksIDs();
+    setCallbacksGraph(adapter);
+    setCallbacksHypergraph(adapter);
+    if (adapter->coordinatesAvailable()) {
+      setCallbacksGeom(adapter->getCoordinateInput());
+    }
+
+    setCallbacksHierTaskMapping_withGraphModel(adapter, machine__);
+  }
+
+
+  // JAE: Not implemented yet
```

Comment:
WHat is not implemented here?  Just the hierarchical part?  Maybe move comment below.